src/bsort.py

```diff
@@ -47,25 +55,46 @@ def count_sort(x: list[int]) -> list[int]:
     >>> count_sort([1, 2, 1, 2, 4])
     [1, 1, 2, 2, 4]
     """
+    # Count-sort where the output list have the final length to begin
+    # with.
     counts = count_keys(x)
-    out = [0] * len(x)
-    # FIXME: do the actual sorting
+    out = [0]*len(x)
+    j = 0
+    for key in range(max(x)+1): # Iterate over all keys. 
+        for i in range(j, counts[key]+j):
+            out[i]=key
+        j += counts[key]
+
+    # Simple count-sort where output list is growing. 
+    #counts = count_keys(x)
+    #out = [] # otherwise out = [0] * len(x)
+    #for key, count in enumerate(counts):
+    #    out.extend([key]*count)
     return out
 
+print('Count sort:')
+print(count_sort([1, 2, 1, 2, 4]))
+
 
 def cumsum(x: list[int]) -> list[int]:
     """
     Calculate the cumulative sum of x.
 
     >>> cumsum([1, 2, 3])
-    [0, 1, 3]
+    [0, 1, 3] # Hvis cumulative sum, burde det ikke være [1, 3, 6]?
```

Comment:
Vi skal have fundet ud af hvor hver nøgle skal starte. Den første nøgle skal altid starte på indeks nul, så du vil altid have et nul først. På det her input har vi talt `[1, 2, 3]` hvilket betyder at vi har en observation med nøgle `0`, to med nøgle `1`, og tre med nøgle `2`. Var de sorterede ville de ligge således:

```
 [0, 1, 1, 2, 2, 2]
```

Den første blok starter på indeks 0 og stopper ved indeks 1, den næste starter på 1 og stopper ved indeks 3, og den sidste starter på 3 og stopper ved 6. Den sidste blok stopper altid ved sidste indeks, så vi behøver ikke regne ud hvor den er.

Det betyder at de indeks vi skal regne ud er

```
 [0, 1, 1, 2, 2, 2]
  0  1      3
```

så vi skal returnere `[0, 1, 3]`.